src/main/java/com/google/devtools/build/lib/rules/proto/ProtoInfo.java

```diff
@@ -37,13 +37,13 @@
 @AutoCodec
 public final class ProtoInfo extends NativeInfo implements ProtoInfoApi<Artifact> {
   /** Provider class for {@link ProtoInfo} objects. */
-  public static class Provider extends BuiltinProvider<ProtoInfo> implements ProtoInfoApi.Provider {
-    public Provider() {
+  public static class ProtoInfoProvider extends BuiltinProvider<ProtoInfo>
```

Comment:
nit: Why this name change? `.Provider` is the usual name for these classes.

Reply:
Looking through the code it's about half `Provider` and half the more specific Provider.  I was just making this match the pattern used by `JavaInfoProvider`

Reply:
/shrug . In that case, i'd lean towards less churn.